opentelekomcloud/resource_opentelekomcloud_cce_node_pool_v3.go

```diff
@@ -0,0 +1,542 @@
+			},
+			NodeManagement: nodepools.NodeManagementSpec{
+				ServerGroupReference: d.Get("server_group_reference").(string),
+			},
+			NodeTemplate: nodes.Spec{
+				Flavor:      d.Get("flavor").(string),
+				Az:          d.Get("availability_zone").(string),
+				Os:          d.Get("os").(string),
+				Login:       loginSpec,
+				RootVolume:  resourceCCERootVolume(d),
+				DataVolumes: resourceCCEDataVolume(d),
+				BillingMode: 0,
+				Count:       1,
+				NodeNicSpec: nodes.NodeNicSpec{
+					PrimaryNic: nodes.PrimaryNic{
+						SubnetId: d.Get("subnet_id").(string),
+					},
+				},
+				ExtendParam: nodes.ExtendParam{
+					PreInstall:  base64PreInstall,
+					PostInstall: base64PostInstall,
+				},
+				Taints:   resourceCCENodeTaints(d),
+				K8sTags:  resourceCCENodeK8sTags(d),
+				UserTags: resourceCCENodePoolUserTags(d),
+			},
+		},
+	}
+
+	clusterId := d.Get("cluster_id").(string)
+	stateCluster := &resource.StateChangeConf{
+		Target:     []string{"Available"},
+		Refresh:    waitForClusterAvailable(nodePoolClient, clusterId),
+		Timeout:    15 * time.Minute,
+		Delay:      15 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+	_, err = stateCluster.WaitForState()
+
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+	s, err := nodepools.Create(nodePoolClient, clusterId, createOpts).Extract()
+	if err != nil {
+		if _, ok := err.(golangsdk.ErrDefault403); ok {
+			retryNode, err := recursiveNodePoolCreate(nodePoolClient, createOpts, clusterId, 403)
+			if err == "fail" {
+				return fmt.Errorf("error creating Open Telekom Cloud Node Pool")
+			}
+			s = retryNode
+		} else {
+			return fmt.Errorf("error creating Open Telekom Cloud Node Pool: %s", err)
+		}
+	}
+
+	if len(s.Metadata.Id) == 0 {
+		return fmt.Errorf("error fetching CreateNodePool id")
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Synchronizing"},
+		Target:       []string{""},
+		Refresh:      waitForCceNodePoolActive(nodePoolClient, clusterId, s.Metadata.Id),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        120 * time.Second,
+		PollInterval: 20 * time.Second,
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error creating Open Telekom Cloud CCE Node Pool: %s", err)
+	}
+
+	d.SetId(s.Metadata.Id)
+	return resourceCCENodePoolV3Read(d, meta)
+}
+
+func resourceCCENodePoolV3Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	nodePoolClient, err := config.cceV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating Open Telekom Cloud CCE Node Pool client: %s", err)
+	}
+	clusterId := d.Get("cluster_id").(string)
+	s, err := nodepools.Get(nodePoolClient, clusterId, d.Id()).Extract()
+	if err != nil {
+		if _, ok := err.(golangsdk.ErrDefault404); ok {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("error retrieving Open Telekom Cloud Node Pool: %s", err)
+	}
+
+	me := &multierror.Error{}
```

Comment:
you can remove initialization and replace with
```go
me := multierror.Append(nil,
		d.Set("name", s.Metadata.Name),
		...
```

Reply:
Even `nil` is not required